src/signals/settings/users/containers/Overview/components/DataView/DataList/__tests__/DataList.test.js

```diff
@@ -0,0 +1,210 @@
+      expect(getByText(String(item.id))).toBeInTheDocument();
+    });
+  });
+
+  it('should render correct number of columns and in correct column order', () => {
+    const alphabetic = [...columns].sort(sortAlphabetic);
+    const alphabeticReversed = [...columns].sort(sortAlphabeticReversed);
+    const expectOrder = (order, negate = false) => (row, rowIDX) => row.childNodes
+      .forEach(
+        (column, idx) => {
+          const expectObj = negate ? expect(column).not : expect(column);
+
+          expectObj.toHaveTextContent(data[rowIDX][order[idx]]);
+        });
+    const { container, rerender } = render(
+      tableWithDataList()
+    );
+
+    let allDataRows = Array.from(container.querySelectorAll('tbody tr'));
+
+    expect(allDataRows).toHaveLength(data.length);
+    allDataRows.forEach(
+      row => expect(row.childNodes).toHaveLength(columns.length)
+    );
+    allDataRows.forEach(expectOrder(alphabetic, true)); // Expect order not to match 'alphabetic'.
+
+    rerender(
+      tableWithDataList({ columnOrder: alphabetic })
+    );
+
+    allDataRows = Array.from(container.querySelectorAll('tbody tr'));
+
+    expect(allDataRows).toHaveLength(data.length);
+    allDataRows.forEach(
+      row => expect(row.childNodes).toHaveLength(columns.length)
+    );
+    allDataRows.forEach(expectOrder(alphabetic));
+
+    rerender(
+      tableWithDataList({ columnOrder: alphabeticReversed })
+    );
+
+    allDataRows = Array.from(container.querySelectorAll('tbody tr'));
+
+    expect(allDataRows).toHaveLength(data.length);
+    allDataRows.forEach(
+      row => expect(row.childNodes).toHaveLength(columns.length)
+    );
+    allDataRows.forEach(expectOrder(alphabeticReversed));
+  });
+
+  it('should not render invisible columns', () => {
+    const invisibleColumns = columns.slice(0, 3);
+    const { container, rerender } = render(
+      tableWithDataList()
+    );
+
+    let allDataRows = Array.from(container.querySelectorAll('tbody tr'));
+
+    expect(allDataRows).toHaveLength(data.length);
+    allDataRows.forEach(
+      row => expect(row.childNodes).toHaveLength(columns.length)
+    );
+
+    rerender(
+      tableWithDataList({ invisibleColumns })
+    );
+
+    allDataRows = Array.from(container.querySelectorAll('tbody tr'));
+
+    expect(allDataRows).toHaveLength(data.length);
+    allDataRows.forEach(
+      row => expect(row.childNodes).toHaveLength(columns.length - invisibleColumns.length)
+    );
+    allDataRows.forEach(
+      (row, rowIDX) => invisibleColumns.forEach(
+        invisibleColumn => {
+          expect(within(row).queryByText(String(data[rowIDX][invisibleColumn]))).toBeNull();
+        }
+      )
+    );
+  });
+
+  it('should render with onItemClick and primaryKeyColumn property', () => {
+    const onItemClickHandler = jest.fn();
+    const { container, rerender } = render(
+      tableWithDataList()
+    );
+
+    expect(onItemClickHandler).toHaveBeenCalledTimes(0);
```

Comment:
Pretty pointless to verify that a handler that hasn't been passed as a prop has not been called, isn't it? ;)